File: src/main/go/easy/minimum_number_of_operations_to_make_elements_in_array_distinct.go
package easy

/*
*
3396. Minimum Number of Operations to Make Elements in Array Distinct
https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/description/
Runtime 0 ms Beats 100.00%
Memory 4.92 MB Beats 67.44%
*/
func minimumOperations(nums []int) int {
	result := 0
	existedNums := [101]int{}
	existedNums[nums[0]] = 1

	for idx := 1; idx < len(nums); idx++ {
		if existedNums[nums[idx]] != 0 {
			tmp := (existedNums[nums[idx]] / 3) + 1
			if result < tmp {
				result = tmp
			}
		}
		existedNums[nums[idx]] = idx
	}

	return result
}

File: src/main/java/ru/femirion/leetcode/easy/MinimumNumberOfOperationsToMakeElementsInArrayDistinct.java
package ru.femirion.leetcode.easy;

/**
 * 3396. Minimum Number of Operations to Make Elements in Array Distinct
 * https://leetcode.com/problems/minimum-number-of-operations-to-make-elements-in-array-distinct/description/
 */
public class MinimumNumberOfOperationsToMakeElementsInArrayDistinct {

    /**
     * Runtime 1 ms Beats 98.39%
     * Memory 44.93 MB Beats 12.90%
     */
    public int minimumOperations(int[] nums) {
        int[] existedNums = new int[101];

        int result = 0;
        for (int i = 0; i < nums.length; i++) {
            if (existedNums[nums[i]] != 0) {
                result = Math.max(result, (existedNums[nums[i]] / 3) + 1);
            }
            existedNums[nums[i]] = i == 0 ? 1 : i;
        }

        return result;
    }
}
